docs/architecture/accounts.md

```diff
@@ -1,211 +1,139 @@
+| **Description**        | For general user accounts (e.g., a wallet) where code changes are allowed. | For typical smart contracts where code is fixed after deployment. | Allows to issue and manage fungible assets. | Allows to issue and manage non-fungible assets. |
+| **Code updatability**  | Yes           | No              | No              | No                  |
+| **Most significant bits** | `00`        | `01`            | `10`            | `11`                |
 
-Currently, Miden provides two standard implementations for account code.
+### The Accounts Storage Mode
 
-##### Basic user account
+Users can choose whether their accounts are stored publicly or privately. The third and fourth most significant bits of the [ID](#id) encode this preference:
 
-There is a standard for a basic user account. It exposes three functions via its interface.
+- **Public Accounts:**  
+  The account’s state is stored on-chain, similar to how accounts are stored in public blockchains like Ethereum. Contracts that rely on a shared, publicly accessible state (e.g., a DEX) should be public.
 
-<details>
-  <summary>Basic user account code</summary>
+- **Private Accounts:**  
+  Only a commitment (hash) to the account’s state is stored on-chain. This mode is suitable for users who prioritize privacy and off-chain data management. To interact with a private account, a user must have knowledge of its interface.
 
-  ```arduino
-    use.miden::contracts::wallets::basic->basic_wallet
-    use.miden::contracts::auth::basic
+## The Accounts Core Components
 
-    export.basic_wallet::receive_asset
-    export.basic_wallet::create_note
-    export.basic_wallet::move_asset_to_note
-    export.basic::auth_tx_rpo_falcon512
-  ```
-</details>
+An `Account` is composed of several core components, illustrated below:
 
-[Note scripts](notes.md#the-note-script) or transaction scripts can call `receive_asset`, `create_note`, and `move_asset_to_note` procedures.
+<p style="text-align: center;">
+    <img src="../img/architecture/account/account-definition.png" style="width:30%;" alt="Account diagram"/>
+</p>
 
-Transaction scripts can also call `auth_tx_rpo_falcon512` to authenticate the transaction.
+These components are:
 
-> **Warning**
-> Without correct authentication, i.e., knowing the correct private key, a note cannot successfully invoke `receive_asset`, `create_note`, or `move_asset_to_note`.
+1. [ID](#id)  
+2. [Storage](#storage)  
+3. [Nonce](#nonce)  
+4. [Vault](#vault)  
+5. [Code](#code)
 
-##### Basic fungible faucet (faucet for fungible assets)
+### ID
 
-There is also a standard for a [basic fungible faucet](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-lib/asm/miden/contracts/faucets/basic_fungible.masm).
+> An immutable and unique identifier for the `Account`.
 
-<details>
-  <summary>Fungible faucet code</summary>
+The `Account` ID is a single field element (`felt`) of about 63 bits. The four most significant bits encode:  
```

Comment:
Is think "single field element" is not needed to understand the Account ID. 

Maybe just say, "A 63-bit long number represents the account ID. It's four most significant bits ..."